tests/unit/cms/classes/ControllerTest.php

```diff
@@ -25,12 +25,12 @@ public function testThemeUrl()
     {
         $theme = Theme::load('test');
         $controller = new Controller($theme);
-
+        
         $url = $controller->themeUrl();
-        $this->assertEquals('http://localhost/themes/test', $url);
+        $this->assertEquals(config('app.url').substr(config('cms.themesPath').'/test', 1), $url);
```

Comment:
Your URL building is off, firstly it's missing a slash at the start and secondly this doesn't properly cover every case of custom configuration. Isn't their a url helper that you can use instead? Something like `url(config('cms.themesPath'))`?

Reply:
I'm not entirely familiar with what all the helper functions are that are available and what the correct way is to build things. This was my first effort to get this one failing unit test to pass and make it useable.

How it builds now is app.url which ends with a / at the end like http://localhost/ or http://127.0.0.1/ when having the default config which needs then the theme added which is stored in config like '/theme' as default, which then needs the '/' at the start removed which is the substr for.

i'll have a look at the possible helper functions tomorrow(now +13h)

Reply:
I modified it with a regex. The url helper doesn't remove double slashes. 
Right now I build the url with a `/` in between that will get removed by the regex if they have a config set up without trailing and leading slash for the corresponding settings. It now passes on several configuration ways.

Reply:
@tschallacka could you instead look at adding that handling to the URL helper? I'd rather the built in helper works for all configurations instead of just one random unit test.

Reply:
Ask and you shall receive. I had to add a few extra files to the project. I hope I put them in the correct places. I'm not sure if i'm supposed to put it in the modules/system or somewhere else. If it needs to be put somewhere else/registered somewhere else please let me know.